pkg/scheduler/nodeinfo/node_info.go

```diff
@@ -665,14 +665,20 @@ func (n *NodeInfo) FilterOutPods(pods []*v1.Pod) []*v1.Pod {
 			continue
 		}
 		// If pod is on the given node, add it to 'filtered' only if it is present in nodeInfo.
-		podKey, _ := GetPodKey(p)
+		var podKey string
+		var err error
+		if podKey, err = GetPodKey(p); err != nil {
+			klog.Errorf("Unable to get pod key: %v", err)
+		}
 		for _, np := range n.Pods() {
-			npodkey, _ := GetPodKey(np)
-			if npodkey == podKey {
-				filtered = append(filtered, p)
-				break
+			if npodkey, err2 := GetPodKey(np); err2 == nil {
```

Comment:
We prefer not to use variable names like `err2`. As long as we are checking the error, we are not concerned about getting them mixed up.